api/v1alpha1/mustgather_types.go

```diff
@@ -47,6 +93,11 @@ type MustGatherSpec struct {
 	// +listType=set
 	MustGatherImages []string `json:"mustGatherImages,omitempty"`
 
+	// The list of must gather images to run with parameters, optional
+	// +kubebuilder:validation:Optional
+	// +listType=set
```

Comment:
According to the kubebuilder documentation, listType of `set`, should be used only for scalar fields.

What do you think about:
```suggestion
	// +listType=map
	// +listMapKey=image
```